### PR TITLE
Landing page improvements

### PR DESCRIPTION
**Pull request by LukeChannings**
I've made a bunch of changes, mostly to the `web` package:

- Landing page: design tweaks
- Landing page: Optimise images
- Landing page: Optimise footer in small viewports (two-column layout)
- Landing page: Improve accessibility by using semantic markup, re-enabling outlines for keyboard focus, and set alt tags where possible.
- Upgrade Next.js to version 13, this enables `next/image` to behave correctly.
- Update TypeScript and React to the minimum versions required by Next 13
- Improve types in numerous places, specifically `useCallback` instances.
- The root font size is set as a percentage instead of hardcoded, as such the application now respects the browser setting for font size.
- Volta: Update workspace projects to extend the root configuration. This prevents a problem where the Node / Yarn version reverts to the default (Yarn 2 being the biggest problem) when changing to a workspace directory.
- Update the licence in package.json to reflect the `LICENCE` file.
- Add a `test:typecheck` script to relevant packages for quick type checking of all packages
- Update the GitHub actions workflow to support running in forked repos

Potential concerns:

- I use Next.js's `<Image />` component, which dynamically optimises images when they are requested. The first time the image is requested this will take some time to render. The rendered sizes are not committed, and will be kept in the Next.js cache. It's important that this cache is persisted, especially in production. See: https://nextjs.org/docs/app/api-reference/components/image#caching-behavior

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -11,10 +11,16 @@ on:
...
+env:
+  NEXT_PUBLIC_APP_ENV: prod
+  NEXT_PUBLIC_BASE_URL: http://localhost:3000
+  NEXT_PUBLIC_SERVER_BASE_URL: http://localhost:4000
+  NEXT_PUBLIC_HIGHLIGHTS_BASE_URL: http://localhost:3000
+
 jobs:
   run-code-tests:
     name: Run Codebase tests
-    runs-on: ubuntu-latest-m
```
Comment (jacksonh) [inline .github/workflows/run-tests.yaml:17]: Do you know why this change was needed? The `-m` should be telling the worker to use the larger sized instances for this job.

File: apple/gql-server/package.json [hunk 2]
```diff
@@ -10,5 +10,8 @@
     "express": "^4.18.1",
     "express-graphql": "^0.12.0",
     "graphql": "^16.8.1"
+  },
+  "volta": {
```
Comment (jacksonh) [inline apple/gql-server/package.json:14]: nice

File: packages/appreader/webpack.config.ts [hunk 3]
```diff
@@ -60,6 +60,11 @@ const config: Configuration = {
   ],
   resolve: {
     extensions: ['.tsx', '.ts', '.js'],
+    fallback: {
```
Comment (jacksonh) [inline packages/appreader/webpack.config.ts:63]: Not sure what this one does?

File: packages/web/components/elements/LabelsPicker.tsx [hunk 4]
```diff
@@ -8,6 +8,11 @@ import { EditLabelChip } from './EditLabelChip'
 import { LabelsDispatcher } from '../../lib/hooks/useSetPageLabels'
 import { EditLabelChipStack } from './EditLabelChipStack'
 
+// AutosizeInput is a Class component, but the types are broken in React 18.
```
Comment (jacksonh) [inline packages/web/components/elements/LabelsPicker.tsx:11]: Yeah in the extension we basically just re-created this functionality fairly easily, so we could probably do the same here.

File: packages/web/components/elements/images/ChromeIcon.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import Image from 'next/image'
 
 export function ChromeIcon(): JSX.Element {
```
Comment (jacksonh) [inline packages/web/components/elements/images/ChromeIcon.tsx:3]: We probably should put real values in here, but I can do that in a separate PR.

File: packages/web/components/patterns/ArticleSubtitle.tsx [hunk 6]
```diff
@@ -96,7 +96,7 @@ export function authoredByText(author: string): string {
 
 export function removeHTMLTags(str: string | null | undefined): string {
   if (typeof str === 'string') {
-    return str.replace(/(<([^>]+)>)/gi, '')
+    return str.replace(/(<([^>]+)>)/gi, '');
```
Comment (jacksonh) [inline packages/web/components/patterns/ArticleSubtitle.tsx:99]: Was this one required? Usually we don't have semicolons

File: packages/web/components/patterns/LibraryCards/LibraryCardStyles.tsx [hunk 7]
```diff
@@ -91,10 +91,10 @@ export const siteName = (
     return ''
   }
   try {
-    return new URL(originalArticleUrl).hostname.replace(/^www\./, '')
+    return new URL(originalArticleUrl).hostname.replace(/^www\./, '');
```
Comment (jacksonh) [inline packages/web/components/patterns/LibraryCards/LibraryCardStyles.tsx:94]: Same question on the semicolons

**Comment by jacksonh** [comment]
Thanks, we do deploy to vercel, so I am guessing next image will be handled properly. In local dev i got this error the first time i ran:

```
upstream image response failed for /_next/static/media/landing-02-newsletters.a9ce2434.png DOMException [AbortError]: The operation was aborted.
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at async invokeRequest (/Users/jacksonh/src/omnivore/node_modules/next/dist/server/lib/server-ipc/invoke-request.js:17:12)
    at async /Users/jacksonh/src/omnivore/node_modules/next/dist/server/next-server.js:365:35
    at async imageOptimizer (/Users/jacksonh/src/omnivore/node_modules/next/dist/server/image-optimizer.js:533:13)
    at async cacheEntry.imageResponseCache.get.incrementalCache (/Users/jacksonh/src/omnivore/node_modules/next/dist/server/next-server.js:519:61)
    at async /Users/jacksonh/src/omnivore/node_modules/next/dist/server/response-cache/index.js:92:36
    at async /Users/jacksonh/src/omnivore/node_modules/next/dist/lib/batcher.js:43:32
    ```

File: .github/workflows/run-tests.yaml (hunk 1 above)
Comment (jacksonh) [reply .github/workflows/run-tests.yaml:17]: Ahh i see this is for forked workflows. I wonder how we can handle this, as the smaller instances can timeout on us in the main repo

File: packages/appreader/webpack.config.ts (hunk 3 above)
Comment (LukeChannings) [reply packages/appreader/webpack.config.ts:63]: It's related to the Next 13 upgrade. `@omnivore/appreader` imports a package from `@omnivore/web` and throws an error unless I fallback the imports.

```
$ webpack --mode production
Webpack Bundle Analyzer saved report to /Users/luke/Developer/omnivore/packages/appreader/build/report.html
assets by status 1.28 MiB [cached] 1 asset
orphan modules 7.36 MiB [orphan] 1521 modules
runtime modules 2.37 KiB 7 modules
cacheable modules 3.17 MiB
  modules by path ../../node_modules/ 1.23 MiB 225 modules
  modules by path ../web/ 240 KiB
    modules by path ../web/node_modules/markdown-it/lib/ 166 KiB 53 modules
    modules by path ../web/styles/*.css 22 KiB 2 modules
    modules by path ../web/node_modules/linkify-it/ 23.9 KiB 2 modules
    + 2 modules
  modules by path ./ 1.71 MiB
    modules by path ./node_modules/style-loader/dist/runtime/*.js 5.75 KiB 6 modules
    modules by path ./node_modules/css-loader/dist/runtime/*.js 2.33 KiB
      ./node_modules/css-loader/dist/runtime/noSourceMaps.js 64 bytes [built] [code generated]
      ./node_modules/css-loader/dist/runtime/api.js 2.26 KiB [built] [code generated]
    ./src/index.jsx + 412 modules 1.7 MiB [built] [code generated]

ERROR in ../../node_modules/next/dist/compiled/gzip-size/index.js 1:2832-2845
Module not found: Error: Can't resolve 'fs' in '/Users/luke/Developer/omnivore/node_modules/next/dist/compiled/gzip-size'
 @ ../../node_modules/next/dist/shared/lib/bloom-filter.js 42:33-77
 @ ../../node_modules/next/dist/shared/lib/router/router.js 1649:36-69
 @ ../../node_modules/next/dist/client/router.js 42:57-95
 @ ../../node_modules/next/router.js 1:0-48
 @ ../web/components/templates/article/SetLabelsControl.tsx 35:0-40 279:15-24
 @ ../web/components/templates/article/SetLabelsModal.tsx 28:0-54 252:32-48
 @ ../web/components/templates/article/SetLabelsModalPresenter.tsx 15:0-50 39:27-41 60:27-41
 @ ../web/components/templates/article/HighlightsLayer.tsx 41:0-77 1106:42-74
 @ ../web/components/templates/article/ArticleContainer.tsx 22:0-74 472:26-41
 @ ./src/index.jsx 12:0-95 108:36-52

ERROR in ../../node_modules/next/dist/compiled/gzip-size/index.js 1:2878-2895
Module not found: Error: Can't resolve 'stream' in '/Users/luke/Developer/omnivore/node_modules/next/dist/compiled/gzip-size'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }
 @ ../../node_modules/next/dist/shared/lib/bloom-filter.js 42:33-77
 @ ../../node_modules/next/dist/shared/lib/router/router.js 1649:36-69
 @ ../../node_modules/next/dist/client/router.js 42:57-95
 @ ../../node_modules/next/router.js 1:0-48
 @ ../web/components/templates/article/SetLabelsControl.tsx 35:0-40 279:15-24
 @ ../web/components/templates/article/SetLabelsModal.tsx 28:0-54 252:32-48
 @ ../web/components/templates/article/SetLabelsModalPresenter.tsx 15:0-50 39:27-41 60:27-41
 @ ../web/components/templates/article/HighlightsLayer.tsx 41:0-77 1106:42-74
 @ ../web/components/templates/article/ArticleContainer.tsx 22:0-74 472:26-41
 @ ./src/index.jsx 12:0-95 108:36-52

ERROR in ../../node_modules/next/dist/compiled/gzip-size/index.js 1:2928-2943
Module not found: Error: Can't resolve 'zlib' in '/Users/luke/Developer/omnivore/node_modules/next/dist/compiled/gzip-size'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "zlib": require.resolve("browserify-zlib") }'
        - install 'browserify-zlib'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "zlib": false }
 @ ../../node_modules/next/dist/shared/lib/bloom-filter.js 42:33-77
 @ ../../node_modules/next/dist/shared/lib/router/router.js 1649:36-69
 @ ../../node_modules/next/dist/client/router.js 42:57-95
 @ ../../node_modules/next/router.js 1:0-48
 @ ../web/components/templates/article/SetLabelsControl.tsx 35:0-40 279:15-24
 @ ../web/components/templates/article/SetLabelsModal.tsx 28:0-54 252:32-48
 @ ../web/components/templates/article/SetLabelsModalPresenter.tsx 15:0-50 39:27-41 60:27-41
 @ ../web/components/templates/article/HighlightsLayer.tsx 41:0-77 1106:42-74
 @ ../web/components/templates/article/ArticleContainer.tsx 22:0-74 472:26-41
 @ ./src/index.jsx 12:0-95 108:36-52

3 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.70.0 compiled with 3 errors in 8008 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command
```

**Comment by jacksonh** [comment]
GOing to deploy this one to our demo environment on vercel to see what happens with the images.

**Comment by jacksonh** [comment]
Getting a few of these on vercel `Error: Missing "key" prop for element in iterator  react/jsx-key` will update in a bit.

File: packages/web/pages/about.tsx [hunk 8]
```diff
@@ -11,7 +11,7 @@ export default function LandingPage(): JSX.Element {
         description="Omnivore is the free, open source, read-it-later app for serious readers."
       />
 
-      <About lang="en" />
+      <About lang="zh" />
```
Comment (jacksonh) [inline packages/web/pages/about.tsx:14]: ok this is the part that changed the about page to Chinese

Comment (LukeChannings) [reply packages/web/pages/about.tsx]: lol, that'll do it. Sorry, must've been when I was trying to figure out how the translations were working (this didn't change it to Chinese locally), and forgot to revert.

**Comment by jacksonh** [comment]
Still trying to figure out why, probably the next or the react upgrade, but it looks like this killed PDF loading. Hoping updating some other libraries fixes it.
